src/core/interfacespec.cpp

```diff
@@ -20,65 +20,55 @@
  *******************************************************************************/
 
 #include "interfacespec.h"
+#include "datatype.h"
 
 USE_STRING_ID(Event);
 
 namespace {
-  TPortId getPortId(CStringDictionary::TStringId paPortNameId,
-                    TPortId paMaxPortNames,
-                    const CStringDictionary::TStringId *paPortNames) {
-    for (TPortId i = 0; i < paMaxPortNames; ++i) {
-      if (paPortNameId == paPortNames[i]) {
-        return i;
-      }
-    }
-    return cgInvalidPortId;
-  }
 
   TPortId getPortId(CStringDictionary::TStringId paPortNameId,
                     std::span<const CStringDictionary::TStringId> paPortNames) {
-    for (TPortId i = 0; i < paPortNames.size(); ++i) {
-      if (paPortNameId == paPortNames[i]) {
-        return i;
-      }
+    auto it = std::find(paPortNames.begin(), paPortNames.end(), paPortNameId);
+    if (it == paPortNames.end()) {
+      return cgInvalidPortId;
     }
-    return cgInvalidPortId;
+    return static_cast<TPortId>(paPortNames.begin() - it);
```

Comment:
I believe that should be:
```suggestion
    return static_cast<TPortId>(it - paPortNames.begin());
```

Reply:
fixed